drivers/display/display_ili9340.c

```diff
@@ -1,154 +1,152 @@
+#endif
 
-	data->spi_config.frequency = DT_INST_0_ILITEK_ILI9340_SPI_MAX_FREQUENCY;
-	data->spi_config.operation = SPI_OP_MODE_MASTER | SPI_WORD_SET(8);
-	data->spi_config.slave = DT_INST_0_ILITEK_ILI9340_BASE_ADDRESS;
+#if (RES_HOR > ILI9XXX_X_MAX) && (RES_HOR > ILI9XXX_Y_MAX)
+#error "Please check height of \"ilitek,ili9340\" device"
+#endif
 
-#ifdef DT_INST_0_ILITEK_ILI9340_CS_GPIOS_CONTROLLER
-	data->cs_ctrl.gpio_dev =
-		device_get_binding(DT_INST_0_ILITEK_ILI9340_CS_GPIOS_CONTROLLER);
-	data->cs_ctrl.gpio_pin = DT_INST_0_ILITEK_ILI9340_CS_GPIOS_PIN;
-	data->cs_ctrl.delay = 0U;
-	data->spi_config.cs = &(data->cs_ctrl);
-#else
-	data->spi_config.cs = NULL;
+#if defined(DT_INST_0_ILITEK_ILI9XXX_X_OFFSET) && \
+	(DT_INST_0_ILITEK_ILI9XXX_X_OFFSET > (ILI9XXX_X_MAX - RES_HOR))
+#error "Please check x-offset of \"ilitek,ili9340\" device"
 #endif
 
-#ifdef DT_INST_0_ILITEK_ILI9340_RESET_GPIOS_CONTROLLER
-	data->reset_gpio =
-		device_get_binding(DT_INST_0_ILITEK_ILI9340_RESET_GPIOS_CONTROLLER);
-	if (data->reset_gpio == NULL) {
-		LOG_ERR("Could not get GPIO port for ILI9340 reset");
-		return -EPERM;
-	}
+#if defined(DT_INST_0_ILITEK_ILI9XXX_Y_OFFSET) && \
+	(DT_INST_0_ILITEK_ILI9XXX_Y_OFFSET > (ILI9XXX_Y_MAX - RES_VER))
+#error "Please check y-offset of \"ilitek,ili9340\" device"
+#endif
 
-	gpio_pin_configure(data->reset_gpio, DT_INST_0_ILITEK_ILI9340_RESET_GPIOS_PIN,
-			   GPIO_OUTPUT_INACTIVE |
-			   DT_INST_0_ILITEK_ILI9340_RESET_GPIOS_FLAGS);
+struct ili9xxx_data {
+#ifdef DT_INST_0_ILITEK_ILI9XXX_RESET_GPIOS_CONTROLLER
+	struct device *rst;
 #endif
+	struct device *cmd;
+	struct device *spi_dev;
+	struct spi_config spi_config;
+#ifdef DT_INST_0_ILITEK_ILI9XXX_CS_GPIOS_CONTROLLER
+	struct spi_cs_control cs_ctrl;
+#endif
+	enum display_pixel_format pixel_format;
+	enum display_orientation orientation;
+	u16_t height;
+	u16_t width;
+	u16_t x_offset;
+	u16_t y_offset;
+};
 
-	data->command_data_gpio =
-		device_get_binding(DT_INST_0_ILITEK_ILI9340_CMD_DATA_GPIOS_CONTROLLER);
-	if (data->command_data_gpio == NULL) {
-		LOG_ERR("Could not get GPIO port for ILI9340 command/data");
-		return -EPERM;
-	}
+static int ili9xxx_transmit(struct ili9xxx_data *data, u8_t cmd, void *tx_data,
+		      size_t tx_len)
+{
+	int rv = 0;
+	struct spi_buf tx_buf = { .buf = &cmd, .len = 1 };
+	struct spi_buf_set tx_bufs = { .buffers = &tx_buf, .count = 1 };
 
-	gpio_pin_configure(data->command_data_gpio, DT_INST_0_ILITEK_ILI9340_CMD_DATA_GPIOS_PIN,
-			   GPIO_OUTPUT |
-			   DT_INST_0_ILITEK_ILI9340_CMD_DATA_GPIOS_FLAGS);
-
-#ifdef DT_INST_0_ILITEK_ILI9340_RESET_GPIOS_CONTROLLER
-	LOG_DBG("Resetting display driver");
-	k_sleep(K_MSEC(1));
-	gpio_pin_set(data->reset_gpio,
-		     DT_INST_0_ILITEK_ILI9340_RESET_GPIOS_PIN, 1);
-	k_sleep(K_MSEC(1));
-	gpio_pin_set(data->reset_gpio,
-		     DT_INST_0_ILITEK_ILI9340_RESET_GPIOS_PIN, 0);
-	k_sleep(K_MSEC(5));
-#endif
+	gpio_pin_set(data->cmd,
+			DT_INST_0_ILITEK_ILI9XXX_CMD_DATA_GPIOS_PIN, 0);
```

Comment:
Can't agree with that. From datasheet:
> The D/CX is the signal of data/command
> selection. When D/CX=’1’, D [17:0] bits are display RAM data or command’s parameters. When D/CX=’0’, D
> [17:0] bits are commands.
But i understand your confusion. variable is called "cmd". I will rename int to data or ncmd (now shure which option is better)

Reply:
No,  `DT_INST_0_ILITEK_ILI9XXX_CMD_DATA_GPIOS_PIN, 1);` is correct because [The D/CX pin of ILI9340 is active low (transmission command byte).](https://github.com/zephyrproject-rtos/zephyr/blob/master/dts/bindings/display/ilitek%2Cili9340.yaml). No need for ncmd or cmd.